src/Analyzers/RoslynExtensions.cs

```diff
@@ -118,7 +118,8 @@ public static IEnumerable<MethodDeclarationSyntax> GetSyntaxNodes(this IMethodSy
     /// <returns>True if the method symbol is contained in the type symbol and has the method name, false otherwise.</returns>
     public static bool IsEqualTo(this IMethodSymbol methodSymbol, INamedTypeSymbol? typeSymbol, string methodName)
     {
-        return methodSymbol.ContainingType.Equals(typeSymbol, SymbolEqualityComparer.Default) &&
+        return (methodSymbol.ContainingType.Equals(typeSymbol, SymbolEqualityComparer.Default) ||
+            methodSymbol.ContainingType.OriginalDefinition.Equals(typeSymbol, SymbolEqualityComparer.Default)) &&
```

Comment:
Can you explain why we made this change?

Reply:
To make both statements to work:
```cs
...
targetMethod.IsEqualTo(this.KnownTypeSymbols.Task, nameof(Task.Run)) ||
targetMethod.IsEqualTo(this.KnownTypeSymbols.TaskT, nameof(Task.Run)) ||
...
```

Not only I need to check if a given type is equal to the provided Symbol as an argument, but also if the type was constructed from a generic definition. Thus, the change.